File: reduks-logger/src/main/kotlin/com/beyondeye/reduks/logger/zjsonpatch/NodeType.java
package com.beyondeye.reduks.logger.zjsonpatch;

import com.google.gson.JsonElement;


class NodeType {
    static final int ARRAY=1;
    static final int OBJECT=2;
//    static final int NULL=3;
    static final int PRIMITIVE_OR_NULL=3;

    public static int getNodeType(final JsonElement node) {
        if(node.isJsonArray()) return ARRAY;
        if(node.isJsonObject()) return OBJECT;
//        if(node.isJsonNull()) return NULL;
        return PRIMITIVE_OR_NULL;
    }
}

File: kjsonpatch-core/src/main/kotlin/com/beyondeye/kjsonpatch/NodeType.kt
package com.beyondeye.kjsonpatch

import com.google.gson.JsonElement


internal object NodeType {
    val ARRAY = 1
    val OBJECT = 2
    //    static final int NULL=3;
    val PRIMITIVE_OR_NULL = 3

    fun getNodeType(node: JsonElement): Int {
        if (node.isJsonArray) return ARRAY
        if (node.isJsonObject) return OBJECT
        //        if(node.isJsonNull()) return NULL;
        return PRIMITIVE_OR_NULL
    }
}
